compaction/compaction_manager.hh

```diff
@@ -352,7 +352,9 @@ private:
     // about invoking it. Ref #10146
     compaction_manager();
 public:
-    compaction_manager(config cfg, abort_source& as);
+    // get_cfg is called on each shard when starting a sharded<compaction_manager>
+    // we need the getter since updateable_value is not shard-safe (#7316)
+    compaction_manager(std::function<config()> get_cfg, abort_source& as);
```

Comment:
Isn't this https://docs.seastar.io/master/classseastar_1_1sharded__parameter.html open-coded?

Reply:
yes.  done in v6